### ISSUE TITLE
Incorrect Caching for Scripts in Razor Components

### ISSUE THREAD

**Issue by TheFanatr** [dotnet/aspnetcore#7183]
# Script Served with Improper Caching Metadata in Default Razor Components Project
Files such as scripts, which are regularly updated during development, are being cached by at least Chrome, preventing changes to be reflected in consecutive runs of the application after changes were made.

## Steps to Reproduce
1. Create a default Razor Components project.
2. Add a `.js` file to `wwwroot` that does something arbitrary like logging `hello world` to the console, then reference it from within the `<body></body>` tag of `index.html`.
3. Build the app, then debug it with Chrome.
4. Stop the app and change the previously-created `.js` file to do something else.
5. Repeat step 3.
6. Observe that Chrome cached the previous version and the new one is not used.

## Temporary Workaround
Invalidate Chrome's cache for the site, then add this to the server code that usually just calls `app.UseStaticFiles()`.
```csharp
app.UseStaticFiles(new StaticFileOptions
{
    OnPrepareResponse = context => 
    {
        if (context.File.PhysicalPath.Contains(@"wwroot\"))
        {
            context.Context.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store";
            context.Context.Response.Headers[HeaderNames.Expires] = "-1";
        }
    }
});
```

## Problem with Temporary Workaround
Ideally, the files should only be downloaded by the browser if they have changed, but the above code forces the files to be downloaded every time.

**Comment by TheFanatr** [comment]
Oddly enough, this seems to only trigger after something specific happens with the project. It does not occur by default, in contrary to what the reproduction steps claim. I think what triggered it was when the page failed to show anything when I added a `@page` directive with `/{message:string}`, which is apparently invalid, something messed up with the cache. Whatever it was, it happened with my test repro, which I did the same message thing to. Attempting to get the same effect to happen again once I have invalidated the cache in Chrome is proving difficult. This may actually just be a side effect of something else, and I erroneously attributed it to default caching behavior.
